### PR TITLE
bruno-cli: init at 1.34.2

### LINKED ISSUE #335270: Package request: bruno-cli
**Project description**
cli for bruno

cli alternative to bruno, the IDE for testing APIs

**Metadata**

* homepage URL: https://docs.usebruno.com/
* source URL: https://github.com/usebruno/bruno/tree/main/packages/bruno-cli
* license: mit
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by gepbird**
Closes https://github.com/NixOS/nixpkgs/issues/335270

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasew** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354111`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bruno-cli</li>
  </ul>
</details>

**Comment by lucasew** [comment]
Em qui., 7 de nov. de 2024 10:44, Lucas Eduardo ***@***.***>
escreveu:

**Comment by lucasew** [comment]
Screenshot_20241107-104832

**Comment by gepbird** [comment]
@lucasew I published the PR before it was working to avoid duplicated work, but thanks for the testing nonetheless!

The `MODULE_NOT_FOUND` error is fixed now, but the packaging needs improvement. Even tho this is a CLI app, removing `cairo`, `pango`, `ELECTRON_SKIP_BINARY_DOWNLOAD = 1;` makes the build fail as `buildNpmPackage` downloads every dependency of all workspaces. I was thinking of using `npmWorkspace = "packages/bruno-cli"`, but that doesn't seem to work, probably because it depends on other workspaces.

File: pkgs/by-name/br/bruno-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  ];
+
+  buildInputs =
+    [
+      pango
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      # fix for: https://github.com/NixOS/nixpkgs/issues/272156
+      apple-sdk_11
+    ];
```
Comment (gepbird) [inline pkgs/by-name/br/bruno-cli/package.nix:30]: @mattpolzin can you check if this necessary? Also a quick test on darwin would be welcome :)

```suggestion
    ++ lib.optionals stdenv.hostPlatform.isDarwin [
      # fix for: https://github.com/NixOS/nixpkgs/issues/272156
      apple-sdk_11
    ];
```

**Comment by gepbird** [comment]
I took some inspiration from https://github.com/NixOS/nixpkgs/pull/354339 and further reduced the closure size from ~200M to ~126M.

File: pkgs/by-name/br/bruno-cli/package.nix (hunk 1 above)
Comment (mattpolzin) [reply pkgs/by-name/br/bruno-cli/package.nix:30]: I can try to find time tonight. A family medical issue has had me away from the computer almost entirely.

Comment (mattpolzin) [reply pkgs/by-name/br/bruno-cli/package.nix:30]: I can confirm that `apple-idk_11` is indeed required here. It is needed to build a native extension for at least one node dependency. granted that may well be a dependency not even needed for the CLI executable (I'm not sure of it), but that sounds like it's an issue with the Bruno CLI build tooling being under-developed so far.

**Comment by mattpolzin** [comment]
Result of `nixpkgs-review pr 354111` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bruno-cli</li>
  </ul>
</details>

**Comment by gepbird** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354111`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bruno-cli</li>
  </ul>
</details>

File: pkgs/by-name/br/bruno-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    npm run build --workspace=packages/bruno-query
+
+    npm run sandbox:bundle-libraries --workspace=packages/bruno-js
+
+    runHook postBuild
+  '';
+
+  npmPackFlags = [ "--ignore-scripts" ];
+
+  preFixup = ''
```
Comment (Aleksanaa) [inline pkgs/by-name/br/bruno-cli/package.nix:56]:
```suggestion
  postInstall = ''
```

File: pkgs/by-name/br/bruno-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      bru --help && touch $out
+    '';
+  };
+
+  meta = {
+    description = "CLI of the open-source IDE For exploring and testing APIs";
+    homepage = "https://www.usebruno.com";
+    license = lib.licenses.mit;
+    mainProgram = "bru";
+    maintainers = with lib.maintainers; [
```
Comment (Aleksanaa) [inline pkgs/by-name/br/bruno-cli/package.nix:90]: bustling 🙃

Comment (gepbird) [reply pkgs/by-name/br/bruno-cli/package.nix:90]: Oh I wanted to ask but forgot to: do you @kashw2 @lucasew @mattpolzin @redyf @water-sucks want to also be a maintainer of the CLI app? Please reply if you don't

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**